python/trifinger_simulation/collision_objects.py

```diff
@@ -145,3 +138,205 @@ def __del__(self):
+            baseOrientation=orientation,
+            baseMass=mass,
+            **self._kwargs,
+        )
+
+        # sets visual properties of the object
+        if color_rgba is not None:
+            self.change_texture(color_rgba=color_rgba)
+
+        # sets dynamics of the object
+        self.change_dynamics(
+            lateral_friction=lateral_friction,
+            spinning_friction=spinning_friction,
+            restitution=restitution,
+        )
+
+    def change_dynamics(
+        self, lateral_friction=1, spinning_friction=0.001, restitution=0
+    ):
+        """Allows to change the dynamics properties of the object.
+
+        Args:
+
+            mass (float): how heavy should this object be
+            lateral_friction (double): the friction coefficient of the object
+            spinning_friction (double): the spinning friction of the object
+                around its contact normal
+            restitution (double): the coefficient of restittuion for the object
+        """
+        pybullet.changeDynamics(
+            bodyUniqueId=self.object,
+            linkIndex=-1,
+            lateralFriction=lateral_friction,
+            spinningFriction=spinning_friction,
+            restitution=restitution,
+            **self._kwargs,
+        )
+
+    def change_texture(self, color_rgba=None, texture_file_path=None):
+        """Changes the texture and color of the object.
+
+        Allows to load some custom image or texture onto the object, for eg.
+        this could be a png file. It also allows to change the object colour.
+        Note that, in case both desired color and desired texture are passed,
+        then both will get loaded onto the object, and the texture will have
+        preference over the color since it will get loaded on top of it.
+
+        Args:
+
+            color_rgba: A list of RGBA values.
+            texture_file_path: Path to a texture that is to be loaded onto
+                the object
+
+        """
+        if texture_file_path is None:
+            if color_rgba is None:
+                raise ValueError(
+                    "This method was called without any values"
+                    "for the desired visual attribute"
+                )
+            else:
+                pybullet.changeVisualShape(
+                    self.object, -1, rgbaColor=color_rgba, **self._kwargs,
+                )
+        else:
+            texture_id = pybullet.loadTexture(texture_file_path)
+            pybullet.changeVisualShape(
+                self.object, -1, textureUniqueId=texture_id, **self._kwargs,
+            )
+
+    def set_state(self, position, orientation=None):
+        """Resets the object state to the provided position and
+        orientation
+
+        Args:
+            position: the position to which the block is to be
+                set
+            orientation: desired to be set
+        """
+        if orientation is None:
+            orientation = [1, 1, 1, 1]
```

Comment:
1. I assume this should be `[0, 0, 0, 1]` (= "no rotation").
2. Better directly use this as default above instead of `None`.